### ISSUE TITLE
updaing env times out golang proxy if lambda is configured to have vpc

### ISSUE THREAD

**Issue by thewillhuang** [apex/up#624]
## Prerequisites

* [x] I am running the latest version. (`up upgrade`)
* [x] I searched to see if the issue already exists.
* [x] I inspected the verbose debug output with the `-v, --verbose` flag.
* [x] Are you an Up Pro subscriber?

## Description
after updating up env, proxy timeout
<img width="1440" alt="screen shot 2018-03-18 at 5 10 13 pm" src="https://user-images.githubusercontent.com/7867215/37572953-ce503c8e-2acf-11e8-874f-8c8ed7513756.png">

## Steps to Reproduce

```js
import express from 'express';
const app = express();
app.listen(process.env.PORT);
console.log(process.env);
```

```bash
$ up env add TEST_ENV=123
```

Describe the steps required to reproduce the issue if applicable.

Join us on Slack https://chat.apex.sh/

**Comment by tj** [comment]
I think this is because it's using `import`, the Lambda has v6.x by default at the moment. If I change it to the following it works:

```js
const express = require('express');
const app = express();
app.listen(process.env.PORT);
console.log(process.env);
```

Although the logs you posted seem a little strange, I get:

```
  Mar 19th 09:13:45am INFO staging 4 waiting for app to listen on PORT
  Mar 19th 09:13:45am ERRO staging 4 /var/task/app.js:1
  Mar 19th 09:13:45am ERRO staging 4 (function (exports, require, module, __filename, __dirname) { import express from 'express';
  Mar 19th 09:13:45am ERRO staging 4                                                               ^^^^^^
  Mar 19th 09:13:45am ERRO staging 4 SyntaxError: Unexpected token import
  Mar 19th 09:13:45am ERRO staging 4     at createScript (vm.js:56:10)
```

AWS mentioned they're trying to improve their speed with Node v8 but for now the only way to get it in there is with a binary https://github.com/apex/up-examples/tree/master/oss/node-8

**Comment by thewillhuang** [comment]
oh I'm using webpack to do the transformations, let me get an example repo up, the only way for me to fix it once it starts happening is to delete the vpc role for the stack, delete the stack, change the project name and re do it, but it seems to be breaking after a few updates for me. whats curious for me is the timeout of 18 seconds, i thought the default was 15, does the proxy have a timeout?

**Comment by tj** [comment]
There's 3 seconds of leeway so that Up has more control over the timeout since API Gateway's hard limit is 30s, but it does look like env is getting stuck from those logs. You're on 0.5.11-pro? I haven't seen this from any version, usually maxes out around 600ms from what I've seen

**Comment by thewillhuang** [comment]
yes im on `0.5.11-pro`

**Comment by thewillhuang** [comment]
right so when it works for me, it usually takes around 100 ms or less, but this happens as soon as i set up vpc or ci or add more env =/

**Comment by tj** [comment]
Ohhhh "up vpc"? Up doesn't support VPC yet, that could be the issue if it can't talk to SSM for the env variables due to some permission issues maybe

**Comment by thewillhuang** [comment]
sorry i ment going into the lambda config to add vpc for the lambda for rds access

**Comment by tj** [comment]
I think that's the problem, I haven't even tried Lambda in a VPC yet personally since I've read a lot of posts about its cold starts being awful (10x slower)

**Comment by thewillhuang** [comment]
got it, thanks for the help. there is no other way to talk to a rds unless i add a VPC to lambda right?

**Comment by tj** [comment]
You could have an RDS that is accessible from the internet and password protect it, I've done that with Lambda in the past, keeping it completely private is definitely more ideal.

I'm not 100% sure but I think you could also define security groups on the RDS to allow access from your Lambda function(s) only, even when it's public/password protected.

**Comment by thewillhuang** [comment]
so another thing is that if you use the envs, every request will be at least 100ms slower?

**Comment by tj** [comment]
Only cold starts, but that should be mitigated with reasonable traffic and Up Pro's new-ish warming feature. I'd like to maybe change it to #547  at some point which would "bake" them in at deploy time, but that would change rollback semantics slightly since it'd revert env vars as well

**Comment by thewillhuang** [comment]
awesome was able to connect to rds and keep the envs working when i did not configure vpc for lambda.

**Comment by dusty** [comment]
I just ran into this same issue.  I have an elasticsearch instance inside a VPC, and I added VPC to my lambda manually.  Everything was working well until I upgraded to pro 0.6.1-pro.  Now my app times out at 18s and appears to be stuck in "initializing secrets".

I've worked around this by moving my client back to the non-pro version and then added a .env file for my secrets and telling up to use that with .upignore.

Does anyone have some ideas on what kind of permissions I may need to setup to get secrets working inside a VPC lambda?  I'll keep digging on it as well, I'm very new to lambda, api gateway, iam, so hints are appreciated.

I also understand its not supported yet in up, was hoping I could get it working manually and still use up for the rest.
